cli.go

```diff
@@ -145,10 +164,35 @@ func NewTGFApplication(args []string) *TGFApplication {
 	kingpin.CommandLine = app.Application
 	kingpin.HelpFlag = app.GetFlag("help-tgf")
 
-	app.Parse(args)
+	_, _ = app.Parse(args)
+	app.TempDirMountLocation = resolveTempMountLocation(temp, tempIsSetByUser, tempLocation, tempLocationIsSetByUser)
 	return &app
 }
 
+// resolveTempMountLocation resolves the mount location based on the
+// deprecated cli options --temp/--no-temp and the new option --temp-location
+// This function will no longer be needed when --temp/--no-temp is deprecated
+// and will be replaced by simply setting the default value for the --temp-location kingpin Flag.
+func resolveTempMountLocation(temp bool, tempIsSetByUser bool, tempLocation MountLocation, tempLocationIsSetByUser bool) MountLocation {
+	// If --temp-location was provided on the command line, it wins
+	if tempLocationIsSetByUser {
+		return tempLocation
+	}
+
+	// If --temp/--no-temp were provided on the command line, use them
+	if tempIsSetByUser {
+		switch temp {
```

Comment:
Why using a switch case on a boolean variable instead of simply do a if?

Reply:
No particular reason.  Just a personal preference on how the code looks.

Reply:
For me, it's about the same of writing:
```go
if (boolVar == true) {
    do this
}
if (boolVar == false) {
    do that
}
```

instead of simply:
```go
if (boolVar) {
    do this
} else {
    do that
}
```